lib/internal/util/iterable-weak-map.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const {
+  Set,
+  SetPrototypeAdd,
+  SetPrototypeDelete,
+  Symbol,
+  WeakMap,
+  WeakMapPrototypeGet,
+  WeakMapPrototypeDelete,
+  WeakMapPrototypeSet,
+} = primordials;
+
+// This class is modified from the example code in the WeakRefs specification:
+// https://github.com/tc39/proposal-weakrefs
+// Licensed under ECMA's MIT-style license, see:
+// https://github.com/tc39/ecma262/blob/master/LICENSE.md
+class IterableWeakMap {
+  #weakMap = new WeakMap();
+  #refSet = new Set();
+  #finalizationGroup = new global.FinalizationRegistry(cleanup);
```

Comment:
@joyeecheung @addaleax when I try to define `FinalizationRegistry` and `WeakRef` in primordials, I get a compilation error because the `WeakRef` and `FinalizationRegistry` are `undefined` at the time when `primordials` is initially loaded.

Any thoughts?

Reply:
WeakRef and FinalizationRegistry cannot be used with snapshots because they can be disabled using a flag at runtime.

Reply:
I think this means we can't use them for a utility like this?

We don't want source maps to stop working because the flag isn't set.

Reply:
It's a V8 `--harmony` flag, that's not supported and undocumented in Node.js. I'd say it's fine to use them.